flexget/plugins/input/gazelle.py

```diff
@@ -0,0 +1,434 @@
+            # Translate the input value to the API value
+            # The str cast converts bools to 'True'/'False' for use as keys
+            # This allows for options that have True/False/Other values
+            return opts[str(val)]
+        elif isinstance(val, list):
+            # Fix yaml parser making a list out of a string
+            return ",".join(val)
+        return val
+
+    def params_from_config(self, config):
+        """Filter params and map config values -> api values"""
+        ret = {}
+        for k, v in config.items():
+            key = self._key(k)
+            if key in self.PARAMS:
+                ret[key] = self._getval(k, v)
+        return ret
+
+    def setup(self, task, config):
+        """Set up a session and log in"""
+        self._session = Session()
+        base_url = config.get('base_url', "").rstrip("/")
+        if base_url:
+            if self.base_url and self.base_url != base_url:
+                log.warning("Using plugin designed for %s on %s - "
+                            "things may break", self.base_url, base_url)
+            self.base_url = base_url
+
+        if not self.base_url:
+            raise PluginError("No 'base_url' configured")
+
+        # The consistent request limiting rule seems to be:
+        # "Refrain from making more than five (5) requests every ten (10) seconds"
+        self._session.add_domain_limiter(TokenBucketLimiter(self.base_url, 2, '2 seconds'))
+
+        # Custom user agent
+        user_agent = config.get('user_agent', "Flexget [Gazelle plugin]")
+        if user_agent:
+            self._session.headers.update({"User-Agent": user_agent})
+        self.username = config['username']
+        self.password = config['password']
+
+        # Login
+        self.authenticate()
+
+        # Logged in successfully, it's ok if nothing matches now
+        task.no_entries_ok = True
+
+
+    def authenticate(self, force=False):
+        """
+        Log in and store auth data from the server
+        Adapted from https://github.com/isaaczafuta/whatapi
+        """
+
+        # clean slate before creating/restoring cookies
+        self._session.cookies.clear()
+
+        if not force:
+            with FlexgetSession() as session:
+                db_session = session.query(GazelleSession).filter(
+                    GazelleSession.base_url == self.base_url,
+                    GazelleSession.username == self.username
+                ).one_or_none()
+                if (db_session and db_session.expires and
+                        db_session.expires >= datetime.utcnow()):
+                    # Found a valid session in the DB - use it
+                    self._session.cookies.update(db_session.cookies)
+                    self.authkey = db_session.authkey
+                    self.passkey = db_session.passkey
+                    log.info("Logged in to %s using cached session", self.base_url)
+                    return
```

Comment:
no need to return if you dont care about return value

Reply:
The return is so the rest of the function isn't executed.

Reply:
right, my bad